### PR TITLE
Dagger to Hilt migration

### PR DESCRIPTION
**Pull request by danfbfern**
**What does this PR do?**

   Migrates DI from dagger to Hilt
   
**Database changed?**

  No

**How should this be manually tested?**

  Flow on how to test this.

**What are the relevant tickets?**

  N/A

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] Database migration (if applicable)?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

**Review by rodolfobcardoso — COMMENTED** [review]
Just want to note that on my side the app is crashing on some screens on my side: navigating to topup screen, when removing a wallet, and in the first iab screen.
Other than that, I think using Hilt for dependency injection instead of Dagger is a great addition to the project since it’s reducing boilerplate code.

**Comment by Muradalsaaideh** [comment]
> Just want to note that on my side the app is crashing on some screens on my side: navigating to topup screen, when removing a wallet, and in the first iab screen. Other than that, I think using Hilt for dependency injection instead of Dagger is a great addition to the project since it’s reducing boilerplate code.

**Comment by Muradalsaaideh** [comment]
https://github.com/Catappult/appcoins-wallet-android/pull/232#pullrequestreview-849646476

File: billing/src/main/java/com/appcoins/wallet/billing/adyen/AdyenPaymentRepository.kt [hunk 1]
```diff
@@ -105,37 +110,37 @@ class AdyenPaymentRepository(private val adyenApi: AdyenApi,
...
 
 
-    @GET("transactions/{uid}")
+    @GET("8.20200815/gateways/adyen_v2/transactions/{uid}")
     fun getTransaction(@Path("uid") uid: String, @Query("wallet.address") walletAddress: String,
                        @Query("wallet.signature")
                        walletSignature: String): Single<TransactionResponse>
 
-    @POST("transactions")
+    @GET("8.20200815/gateways/adyen_v2/transactions")
```
Comment (rodolfobcardoso) [inline billing/src/main/java/com/appcoins/wallet/billing/adyen/AdyenPaymentRepository.kt]: This requests need to be a post method to work

File: billing/src/main/java/com/appcoins/wallet/billing/adyen/AdyenPaymentRepository.kt [hunk 2]
```diff
@@ -105,37 +110,37 @@ class AdyenPaymentRepository(private val adyenApi: AdyenApi,
...
     fun makePayment(@Query("wallet.address") walletAddress: String,
                     @Query("wallet.signature") walletSignature: String,
                     @Body payment: Payment): Single<AdyenTransactionResponse>
 
     @Headers("Content-Type: application/json;format=product_token")
-    @POST("transactions")
+    @GET("8.20200815/gateways/adyen_v2/transactions")
```
Comment (rodolfobcardoso) [inline billing/src/main/java/com/appcoins/wallet/billing/adyen/AdyenPaymentRepository.kt]: This requests need to be a post method to work

**Review by rodolfobcardoso — COMMENTED** [review]
The last changes fixed those 3 crashes.
Credit card and paypal requests are not working though. I left a comment where I think these problems are coming from

**Pull request merged**